cmd/syncthing/gui.go

```diff
@@ -969,12 +970,27 @@ func (s *apiService) getRandomString(w http.ResponseWriter, r *http.Request) {
 func (s *apiService) getDBIgnores(w http.ResponseWriter, r *http.Request) {
 	qs := r.URL.Query()
 
-	ignores, patterns, err := s.model.GetIgnores(qs.Get("folder"))
-	if err != nil {
-		http.Error(w, err.Error(), 500)
+	folderName := qs.Get("folder")
+
+	l.Infoln("DBIgnores: folders:", s.cfg.Folders())
+	folder, ok := s.cfg.Folders()[folderName]
+	if !ok {
+		http.Error(w, "folder does not exist", 500)
 		return
 	}
 
+	ignores, patterns, err := s.model.GetIgnores(folderName)
+	if err != nil {
+		matcher := ignore.New(false)
+		path := filepath.Join(folder.Path(), ".stignore")
+		if err := matcher.Load(path); err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
+		ignores = matcher.Lines()
+		patterns = matcher.Patterns()
+	}
```

Comment:
Maybe we should move this to the model (or make this the default implementation in the model), so that there are is a single way to get ignores?